### PR TITLE
itertools.pyi: Add overload to Python 3

### LINKED ISSUE #2561: itertools.product with no arguments support
It is [documented](https://docs.python.org/3/library/itertools.html#itertools.product) that `itertools.product` call with no arguments works fine and it is:

**test.py**
```python
from itertools import product

print(next(product()))  # ()
```

tested on *Python 3.3+* on *Linux* and on *Python 3.0+* on *Windows*, but running `mypy` raises error
```
> mypy test.py
test.py:3: error: All overload variants of "product" require at least one argument
test.py:3: note: Possible overload variants:
test.py:3: note:     def [_T1] product(iter1: Iterable[_T1]) -> Iterator[Tuple[_T1]]
test.py:3: note:     def [_T1, _T2] product(iter1: Iterable[_T1], iter2: Iterable[_T2]) -> Iterator[Tuple[_T1, _T2]]
test.py:3: note:     <6 more similar overloads not shown, out of 8 total overloads>
```
it happens because there is no support for this kind of call in `typeshed`, can be fixed by specifying that `repeat` parameter (added in [this PR](https://github.com/python/typeshed/pull/1410)) has default argument like
```python
@overload
def product(*iterables: Iterable[Any], repeat: int = ...) -> Iterator[Tuple[Any, ...]]: ...
```

### PR DESCRIPTION
**Pull request by utkarsh2102**
This adds an overload to Python 3 version which was missing.

Fixes #2561

### CODE REVIEW

**Comment by utkarsh2102** [comment]
I'm not sure if this would be the actual solution to the issue.
Because there seems to be return type issue.
I guess the CI would fail.

File: stdlib/3/itertools.pyi [hunk 1]
```diff
@@ -97,6 +97,8 @@ def product(iter1: Iterable[Any],
             *iterables: Iterable[Any]) -> Iterator[Tuple[Any, ...]]: ...
 @overload
 def product(*iterables: Iterable[Any], repeat: int) -> Iterator[Tuple[Any, ...]]: ...
```
Comment (JelleZijlstra) [inline stdlib/3/itertools.pyi]: Does just adding `= ...` to `repeat` here work? I feel like this and the next overload would be overlapping.

**Comment by utkarsh2102** [comment]
So although the CI is green now, but is it the __right__ way?

**Comment by srittau** [comment]
Unfortunately we lose the information that an empty tuple is returned when no iterables are passed in, but this is probably the best we can do.

**Pull request merged**